### PR TITLE
Gracefully handle non-existant target files in brain::Circuit, other cleanups

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
ping

File: brain/detail/circuit.h [hunk 1]
```diff
@@ -108,9 +108,20 @@ std::string toString( const std::string& in ) { return in; }
...
 
-typedef boost::unordered_map< std::string, neuron::MorphologyPtr > Loaded;
+template<typename T>
+void _shuffle( T& container )
+{
+    std::random_device randomDevice;
+    std::mt19937_64 randomEngine( randomDevice( ));
+    const char* seedEnv = getenv( "BRAIN_CIRCUIT_SEED" );
+    if( seedEnv )
+        randomEngine.seed( std::stoul( seedEnv ));
```
Comment (hernando) [inline brain/detail/circuit.h]: Can stoul throw?

**Comment by hernando** [comment]
+1 otherwise

File: brain/detail/circuit.h (hunk 1 above)
Comment (tribal-tec) [inline brain/detail/circuit.h]: Yes, addressed.

**Comment by hernando** [comment]
Retest this please.

**Comment by hernando** [comment]
Retest this please

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
retest this please

**Pull request merged**